src/main/java/gregtech/common/pipelike/itempipe/net/ItemNetHandler.java

```diff
@@ -293,7 +293,9 @@ public ItemStack insert(ItemPipeNet.Inventory handler, ItemStack stack, boolean
 
     public ItemStack insert(ItemPipeNet.Inventory handler, ItemStack stack, boolean simulate, boolean ignoreLimit) {
         int allowed = ignoreLimit ? stack.getCount() : checkTransferable(handler.getProperties().getTransferRate(), stack.getCount(), simulate);
-        if (allowed == 0) return stack;
+        if (allowed == 0 || !handler.matchesFilters(stack)) {
+            return stack;
```

Comment:
A `Inventory` stores only a single `IItemHandler` position. It also stores all filters that it collected on the way during walk. So it will only check filters that were on the shortest path to the Inventory.